packages/resources/src/Function.ts

```diff
@@ -373,6 +373,10 @@ export class Function extends lambda.Function {
     } as FunctionHandlerProps);
 
     this._isLiveDevEnabled = isLiveDevEnabled;
+    
+    // Fixes CDK issue with Provisioned Concurrency and Lambda Versions
+    // cf. https://github.com/aws/aws-cdk/issues/13731#issuecomment-814801449 
+    this.currentVersion;
```

Comment:
Don't think we should always calling `this.currentVersion`. @thdxr can you confirm on this?